scully/utils/defaultAction.ts

```diff
@@ -54,11 +58,61 @@ async function doChunks(dataRoutes) {
     const x = await acc;
     const activeChunk = await Promise.all(
       part.map(async (handledRoute: HandledRoute) =>
-        routeContentRenderer(handledRoute).then((html: string) =>
-          writeToFs(handledRoute.route, html)
-        )
+        routeContentRenderer(handledRoute).then((html: string) => writeToFs(handledRoute.route, html))
       )
     );
     return x.concat(activeChunk);
   }, Promise.resolve([]));
 }
+
+async function addExtraRoutes(config): Promise<string[]> {
+  let extraRoutes = config.extraRoutes;
```

Comment:
@SanderElias please read this now. I changed it to allow you to pass a hodge-podge array of string and promises, and it will take care of resolving everything, removing empty values, and flattening nested arrays of unhandled routes. 

```javascript 
[ '/user/:id', Promise<string>, Promise<string[]>, undefined]
```
 gets turned into 
```javascript
['/user/:id', '/foo/:id', '/bar/:id', '/baz/:id']
```

Reply:
awsome, that is even more as I would have wanted ;)